unittests/CMakeLists.txt

```diff
@@ -7,6 +7,16 @@
 #
 #===------------------------------------------------------------------------===#
 
+if (TARGET gtest OR TARGET gtest_main)
```

Comment:
This test should be moved after `set(GTEST_SRC_DIR`.
It should be possible to provide a gtest version different than the version provided with LLVM.
And only if none was found, the LLVM-based version should be used as a backup and the user warned about it
if a different path has been set before.

Reply:
While I agree that this would be nicer, this leads to exactly the problem that I address with this PR, i.e. the custom version of gtest tries to register `gtest` and `gtest_main` as targets, which fails:
```
CMake Error at /path/to/custom/googletest/googletest/cmake/internal_utils.cmake:149 (add_library):
  add_library cannot create target "gtest" because an imported target with
  the same name already exists.
Call Stack (most recent call first):
  /path/to/custom/googletest/googletest/cmake/internal_utils.cmake:172 (cxx_library_with_type)
  /path/to/custom/googletest/googletest/CMakeLists.txt:90 (cxx_library)
[the same for `gtest_main`]
```
Unless you know of a way to get rid of an imported target (I did not find one), I fear we will have to live with this limitation.

Reply:
Note that this only happens if the LLVM source tree is available and when using LLVM with `find_package()` instead of `llvm-config`. I check for ` if (TARGET gtest OR TARGET gtest_main) ` first to only reuse googletest when absolutely necessary (LLVM's googletest might also be reused with `llvm-config`, but this PR does not do anything about that, so you would have to manually specify the path).